pandas/core/indexes/datetimes.py

```diff
@@ -174,8 +174,92 @@ def _new_DatetimeIndex(cls, d):
     return result
 
 
-class DatetimeIndex(DatelikeOps, TimelikeOps, DatetimeIndexOpsMixin,
-                    Int64Index):
+class DatetimeArray(DatetimeLikeArray):
```

Comment:
I don't think we've discussed this anywhere yet, but I'm not sure if we want a plain `DatetimeArray`, just a `DatetimeTZArray`. We'll need to hash that out somewhere. That discussion probably depends on how public these EAs are going to be.

Reply:
nump's impl has served us, but we it is immensely easier to have a *real* DTI be the actual underlying implementation as we can easily extend this. So I am onboard with @jbrockmendel here to have a combined DatetimeArray. We actually discussed this I think in 0.17.0 when I created this originally, but was rejected for compat with numpy. We could still have that (the issue is what ``.values`` outputs). but it makes the code much better to have this than not.

Reply:
@jreback glad to hear you're on board.  Any thoughts on the appropriate size/scope per PR to make reviewers' task easier?

Reply:
i think could move much of this to core/array/datetime.py

doing straight moves first then changes are good